src/uct/base/uct_iface.c

```diff
@@ -222,8 +222,18 @@ int uct_iface_is_reachable(const uct_iface_h iface, const uct_device_addr_t *dev
 int uct_iface_is_reachable_v2(const uct_iface_h iface,
                               const uct_iface_is_reachable_params_t *params)
 {
-    ucs_fatal("uct_iface_is_reachable_v2 not supported yet");
-    return 0;
+    const uct_base_iface_t *base_iface = ucs_derived_of(iface, uct_base_iface_t);
+
+    UCT_CHECK_PARAM(params->field_mask & UCT_IFACE_IS_REACHABLE_FIELD_DEVICE_ADDR &
+                                         UCT_IFACE_IS_REACHABLE_FIELD_IFACE_ADDR &
+                                         UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING &
+                                         UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING_LENGTH,
```

Comment:
Oops! My fault. It would be splited to:
```
    UCT_CHECK_PARAM(params->field_mask & UCT_IFACE_IS_REACHABLE_FIELD_DEVICE_ADDR,
                    "UCT_IFACE_IS_REACHABLE_FIELD_DEVICE_ADDR is not defined");
    
    UCT_CHECK_PARAM(params->field_mask & UCT_IFACE_IS_REACHABLE_FIELD_IFACE_ADDR,
                    "UCT_IFACE_IS_REACHABLE_FIELD_IFACE_ADDR is not defined");

    UCT_CHECK_PARAM(params->field_mask & UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING,
                    "UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING is not defined");
    
    UCT_CHECK_PARAM(params->field_mask & UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING_LENGTH,
                    "UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING_LENGTH is not defined");
```

Reply:
I don't think that all fields are mandatory for all transports, I would expect here checking of values in case of flags are present. For example, info_string is not NULL if UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING and its lenght is also set and > 0

Reply:
Sure. How about change it to:
```
    if ((params->field_mask & UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING) ||
        (params->field_mask & UCT_IFACE_IS_REACHABLE_FIELD_INFO_STRING_LENGTH)) {
            UCT_CHECK_PARAM((params->info_string != NULL),
                    "info_string shouldn't be null");
            UCT_CHECK_PARAM((params->info_string_length > 0),
                    "info_string_length should be greater than zero");
    }
```
Thanks